android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/CreateEngineSessionMiddleware.kt

```diff
@@ -88,16 +88,19 @@ private fun getOrCreateEngineSession(
     val tab = store.state.findTabOrCustomTab(tabId)
     if (tab == null) {
         logger.warn("Requested engine session for tab. But tab does not exist. ($tabId)")
+        store.dispatch(EngineAction.UpdateEngineSessionInitializingAction(tabId, false))
```

Comment:
drive by: could we abstract this Action dispatch to a private function with a name that represents what we're doing here? That way the (maybe speculative) fix for this can have some descriptive context and be better understood by future readers of this block.

something akin to:

```
private fun Store.resetEngineInitializationFlag(...) {
    dispatch(EngineAction.UpdateEngineSessionInitializingAction(tabId, false))
}
```